proto/api/v1/central_events_service.proto

```diff
@@ -0,0 +1,143 @@
+syntax = "proto3";
+
+option go_package = "v1";
+
+option java_package = "io.stackrox.proto.api.v1";
+
+import weak "google/api/annotations.proto";
+import "api/v1/common.proto";
+import "api/v1/pagination.proto";
+import "google/protobuf/timestamp.proto";
+
+package v1;
+
+// CentralEvents are emitted by Central. They are used to create transparency for users
+// for asynchronous, background tasks. events are part of Central's system health view.
+message CentralEvent {
+    // UUID of the event.
+    string id                                  = 1;
+
+    // Type of the event.
+    CentralEventType type                      = 2;
+
+    // Level associated with the event. The level is categorized into danger, warn, info,
+    // success.
+    CentralEventLevel level                    = 3;
+
+    // Message associated with the event. The message may include detailed information
+    // for this particular event.
+    string message                             = 4;
+
+    // Hint associated with the event. The hint may include different information based
+    // on the type of event. It can include instructions to resolve a event, or
+    // informational hints.
+    string hint                                = 5;
+
+    // Domain associated with the event. A event's domain outlines the feature domain where
+    // the event was created from. As an example, this might be "Image Scanning".
+    // In case of events that cannot be tied to a specific domain, this will be "General".
+    string domain                              = 6;
+
+    // Resource type associated with the event. A event may refer to an underlying resource
+    // such as a particular image. In that case, the resource type will be filled here.
+    string resource_type                       = 7;
+
+    // Resource ID associated with the event. If a event refers to an underlying resource,
+    // the resource ID identifies the underlying resource.
+    string resource_id                         = 8;
+
+    // Occurrences associated with the event. When events may occur multiple times, the
+    // occurrences track the amount.
+    int64 num_occurrences                      = 9;
+
+    // Specifies the time when the event has last occurred.
+    google.protobuf.Timestamp last_occurred_at = 10;
+
+    // Specifies the time when the event has been created.
+    google.protobuf.Timestamp created_at       = 11;
+}
+
+// CentralEventType exposes the different types of events.
+enum CentralEventType {
+    CENTRAL_EVENT_TYPE_UNKNOWN     = 0;
+    CENTRAL_EVENT_TYPE_GENERIC     = 1;
+    CENTRAL_EVENT_TYPE_LOG_MESSAGE = 2;
+}
+
+// CentralEventLevel exposes the different levels of events.
+enum CentralEventLevel {
+    CENTRAL_EVENT_LEVEL_UNKNOWN = 0;
+    CENTRAL_EVENT_LEVEL_INFO    = 1;
+    CENTRAL_EVENT_LEVEL_SUCCESS = 2;
+    CENTRAL_EVENT_LEVEL_WARN    = 3;
```

Comment:
Is **WARN** a widespread backend convention? A question, not necessarily a change request, because frontend code will need to map anyway.

It is a verb compared to UNKNOWN adjective and INFO SUCCESS DANGER nouns.

For example, PatternFly has:
* default (teal bell)
* info (blue circle info)
* success (green check mark)
* warning (yellow triangle exclamation)
* danger (red circle exclamation)

I do see **WARN** versus **INFO** results in global search related to messages.

Reply:
Yeah, I think we can use `WARNING` instead. I suppose we sometimes use the short version in protos / on the backend, but `WARNING` is definitely more in-line with the patternfly stuff.

Reply:
It might be more intuitive for API to use `CENTRAL_EVENT_LEVEL_ERROR` even if the presentation in PatternFly is called danger.

Reply:
Changed this to error as per your suggestion.